### PR TITLE
Fix warnings that will be treated as errors in CMSSW

### PR DESCRIPTION
**Pull request by ariostas**
I fixed a bunch of warnings that are treated as errors in CMSSW by default. The main thing was fixing comparisons between signed and unsigned integers. I changed some variable to be unsigned integers, since it makes more sense for them to be unsigned. I also had to remove a bunch of unused variables. I was undecided between deleting the lines or commenting them out. I decided on deleting them since they looked like remnants from old code.

I also removed the dependency on cppitertools, which was a very minor change that I don't think necessitates its own PR.

There are still a lot of warnings remaining. I think it is definitely worth taking a look at them since some of them look important. However, this PR is already somewhat lengthy, so maybe I'll do that in a later PR.

Also, I noticed that there is a lot of repeated code. I think that it would be worth doing some refactoring at some point to make things more modular.

### CODE REVIEW

File: SDL/MiniDoublet.h [hunk 1]
```diff
@@ -13,8 +13,8 @@ namespace SDL {
     unsigned int* anchorHitIndices;
     unsigned int* outerHitIndices;
     uint16_t* moduleIndices;
-    int* nMDs;             //counter per module
-    int* totOccupancyMDs;  //counter per module
+    unsigned int* nMDs;             //counter per module
+    unsigned int* totOccupancyMDs;  //counter per module
```
[the remark this answers is not in the document]
Comment (slava77) [reply SDL/MiniDoublet.h:17]: > I was thinking if it's more practical to switch to explicit size integer types.

this is outside of this PR scope

**Comment by slava77** [comment]
@ariostas 
when you are done, please run a (standalone) test

File: SDL/MiniDoublet.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (ariostas) [reply SDL/MiniDoublet.h:17]: that's true, I'll think about how much work that would be

**Review by slava77 — APPROVED** [review]

**Pull request merged**